### DISCUSSION TITLE
Running periodic code on server?

### DISCUSSION THREAD

**Discussion by dajack05** [nuxt/nuxt#10667] [General]
Hi all,

I tried asking this question in the "Help" category but didn't get any bites, so I thought I'd ask here.

What is the standard/correct way to run functions on the server which run periodically (say once a day)? My application needs to review data and send emails once a day but I don't see a clear way to do that in Nuxt.

I apologize if I'm missing something obvious.

Thanks!
-
Daniel J

**Comment by rguttersohn** [comment]
I don't think that's a Nuxt issue. You probably need to add some sort of cron job. I went to NPM and found this module for node: [https://www.npmjs.com/package/node-cron](url).

Will that do?

**Reply by dajack05** [reply]
Something along those lines could work. I agree it's not an "issue" with Nuxt, I'm just trying to see how others have solved similar problems so I don't go down a bad route lol

With an ExpressJS backend, I'd just throw a setInterval in the main file which runs once a day, I guess I assumed something like that could exist in a Nuxt project.

**Reply by dajack05** [reply]
Plus, wouldn't node-cron run on the client-side? I don't need 1000 corn jobs checking every day at noon, I just need one job to run each day.

**Reply by rguttersohn** [reply]
Hmmm ... why would it run on the client side? Just read over the documentation for the module, and I don't see anything indicating it would run it on the frontend.

**Reply by dajack05** [reply]
If it were used in a component, that would be client-side, wouldn't it? And I don't see a clear server-side place to put it. I could be missing something basic though.

**Reply by rguttersohn** [reply]
If you add a server folder to your root, you can add the server-side api, routes, and middleware subdirectories. That's where your Node.js will live.

**Reply by rguttersohn** [reply]
[Here is the current documentation for the server directory. ](https://v3.nuxtjs.org/guide/directory-structure/server/) So I don't blame you for missing it.

**Reply by dajack05** [reply]
So is there a way to have something server-side run at startup? Otherwise, I'd have to make something like `/api/startup` and trigger that manually to start the node-cron stuff.

**Reply by rguttersohn** [reply]
I see your dilemma. What if you made a file in the server folder called cron-job and added it there? Have it console.log something every minute or so and see if it does it on its own or needs a request to get things up and running. I've honestly never written one before, so I am basically guessing.

**Reply by dajack05** [reply]
I'll give that a shot and see what happens!

**Reply by rguttersohn** [reply]
Great. Let me know what happens if you don't mind. You might need to register it in a custom subdirectory. I am not sure.

**Reply by dajack05** [reply]
No luck...

Worst case, I could have a separate file who's only job is to poke the server once in a while, then run that alongside the main nuxt app. Just less than ideal.

**Reply by dajack05** [reply]
So, after a bunch of digging, I think I can do what I want using a module.

I created a file called "modules/startup.ts" and added it to the nuxt.config and it seems to work so far.

```js
// nuxt config
export default defineNuxtConfig({
  modules: [  './modules/startup'  ],
});
```

```js
// modules/startup.ts
export default async (inlineOptions: any, nuxt: Nuxt) => {
  nuxt.hook("ready", async (nuxt) => {
    cron.schedule("* * * * * *", (now)=>{
      console.log("Hello From Cron");
    });
  });
};
```
